packages/inferno-component/src/index.ts

```diff
@@ -91,7 +91,7 @@ function queueStateChanges<P, S>(component: Component<P, S>, newState, callback:
 			addToQueue(component, false, callback);
 		}
 	} else {
-		component.state = Object.assign({}, component.state, component._pendingState);
+		component.state = Object.assign(component.state, component._pendingState, newState);
```

Comment:
We can do away with the assignment altogether here to be just `Object.assign(component.state, component._pendingState, newState);`